src/model/mixin/itemStyle.ts

```diff
@@ -35,7 +35,8 @@ export const ITEM_STYLE_KEY_MAP = [
     ['lineDashOffset', 'borderDashOffset'],
     ['lineCap', 'borderCap'],
     ['lineJoin', 'borderJoin'],
-    ['miterLimit', 'borderMiterLimit']
+    ['miterLimit', 'borderMiterLimit'],
+    ['decal']
```

Comment:
This underlying method maps the `itemStyle` in echarts to the style object in zrender.

But `decal` in echarts is different with `decal` in zrender. In echarts it's an object that describe how the decal looks like. But in zrender it's a image that rendered as [pattern](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern) on top of the element.  So we didn't include `decal` here directly. Instead the decal property is created in https://github.com/apache/echarts/blob/4a52199842039d7fa6019ee8584d1a395fbc771a/src/visual/decal.ts#L44

Reply:
Thank you for pointing it out. I agree I should use it to deal with interface `decal`. But the problem is that I cannot find other function than https://github.com/apache/echarts/blob/4a52199842039d7fa6019ee8584d1a395fbc771a/src/component/legend/LegendView.ts#L565
But `getItemStyle()` will only pick the attributes in the list of ITEM_STYLE_KEY_MAP. The only other way I think is to directly visit the decal attribute level by level, which I think is not elegant enough. And it could be risky if we call a structure like `legendItemModel.ecModel.option.legend[0].itemStyle.decal`, because if any structure in between does not exist or has changed, it could result in unstability. Is there any good one to pick the attribute `decal` without adding it to the list of ITEM_STYLE_KEY_MAP?

Reply:
Using `Model#get` or `Model#getShallow` is the best way to access the property in the option. `getItemStyle` is also using `Model#getShallow` at the bottom.

https://github.com/apache/echarts/blob/f9bd8da1884492fdfa234a95df86ff5605708bda/src/model/mixin/makeStyleMapper.ts#L46

These two method works like optional chaining operator. Some examples:
```ts
// Get the itemStyle object in series.
seriesModel.getShallow('itemStyle')
// Get itemStyle.color value in series. Similar to seriesModel.option?.itemStyle?.color
seriesModel.getShallow(['itemStyle', 'color'])
```

The difference between `Model#get` and `Model#getShallow` is `Model#get` will also consider cascading. If we didn't find `itemStyle.color` in the data level, it will keep finding in the series level and global option level.

There is another method `Model#getModel` will wrap a new model after getting the value:

```
const itemStyleModel = seriesModel.getModel('itemStyle');
// itemStyleModel is now in the itemStyle scope
const color = itemStyleModel.get('color');
```

Reply:
Thank you for your detailed explanation! My appreciate for your patience.
Now I use `getShallow` to get `decal` and use `createOrUpdatePatternFromDecal` to deal with it. But one last thing is to get `api` for `createOrUpdatePatternFromDecal`, which I found to be at `legendItemModel.ecModel.scheduler.api`. Is there a better way to access it?

Reply:
`api` will be given in the `render` method 
https://github.com/apache/echarts/blob/72f4c419ae59e738740987b0d6affde96bca7f4b/src/component/legend/LegendView.ts#L100
You can pass it down to other methods if the method needs it.

BTW: I was wrong about `Model#get` and `Model#getShallow`. Just updated in the previous comment. `get(path, false)` in your commited code is the way to ignore option cascading.

Reply:
Just updated the code. I think I used `getShallow` in code and as I track the dataflow, it traces back to parent model if option is not found. So I think it works well in the code.